src/ORM/Hierarchy/MarkedSet.php

```diff
@@ -47,12 +48,12 @@ class MarkedSet
     public $markingFilter;
 
     /**
-     * @var DataObject
+     * @var DataObject&Hierarchy
```

Comment:
There's a check in the constructor that the DataObject that's assigned to $this->rootNode has the Hierarchy extension, and there's no other assignements to this property.